ui/kotlinx-coroutines-android/r8-test-rules.pro

```diff
@@ -0,0 +1,19 @@
+# Entry point for retaining MainDispatcherLoader which uses a ServiceLoader.
+-keep class kotlinx.coroutines.Dispatchers {
+  ** getMain();
+}
+
+# Entry point for retaining CoroutineExceptionHandlerImpl.handlers which uses a ServiceLoader.
+-keep class kotlinx.coroutines.CoroutineExceptionHandlerKt {
+  void handleCoroutineException(...);
+}
+
+# Ensure the custom, fast service loader implementation is removed. In the case of fast service
+# loader encountering an exception it falls back to regular ServiceLoader in a way that cannot be
+# optimized out by R8.
+-include resources/META-INF/com.android.tools/r8-min-1.6.0/coroutines.pro
+-checkdiscard class kotlinx.coroutines.internal.FastServiceLoader
+
+# We are cheating a bit by not having android.jar on R8's library classpath. Ignore those warnings.
+-ignorewarnings
```

Comment:
It seems that the warning are still printed on the console during build. Is there any way to suppress them?

Reply:
I don't think so, unfortunately. If I disable the ErrorStream for the task we will lose the actual errors that we care about in case of failure.